e2etests/tests/unnumbered.go

```diff
@@ -0,0 +1,204 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package tests
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"time"
+
+	frrk8sv1beta1 "github.com/metallb/frr-k8s/api/v1beta1"
+	"github.com/metallb/frrk8stests/pkg/config"
+	"github.com/metallb/frrk8stests/pkg/dump"
+	"github.com/metallb/frrk8stests/pkg/infra"
+	"github.com/metallb/frrk8stests/pkg/k8s"
+	"github.com/metallb/frrk8stests/pkg/k8sclient"
+	"github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.universe.tf/e2etest/pkg/frr"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+
+	frrcontainer "go.universe.tf/e2etest/pkg/frr/container"
+)
+
+var FRRImage string
+
+// The Unnumbered test can run on IPv4, IPv6 or Dual because it needs only the
+// IPv6 LLA addresses in the interface, not full IPv6 stack.
+var _ = ginkgo.Describe("Unnumbered configure BGP peering", func() {
+	var (
+		nodeWithP2PConnection  corev1.Node
+		updater                *config.Updater
+		unnumberedExternalPeer *frrcontainer.FRR
+		cs                     clientset.Interface
+	)
+
+	ginkgo.BeforeEach(func() {
+		if _, found := os.LookupEnv("RUN_FRR_CONTAINER_ON_HOST_NETWORK"); found {
+			ginkgo.Skip("Skipping this test because RUN_FRR_CONTAINER_ON_HOST_NETWORK is set to true")
+		}
+		var err error
+		updater, err = config.NewUpdater()
+		Expect(err).NotTo(HaveOccurred())
+
+		cs = k8sclient.New()
+		nodes, err := k8s.Nodes(cs)
+		Expect(err).NotTo(HaveOccurred())
+		nodeWithP2PConnection = nodes[0]
+
+		unnumberedExternalPeer, err = frrcontainer.SetupP2PPeer(FRRImage, nodeWithP2PConnection)
+		Expect(err).NotTo(HaveOccurred())
+		ginkgo.DeferCleanup(func() {
+			err := frrcontainer.Delete([]*frrcontainer.FRR{unnumberedExternalPeer})
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		err = updater.Clean()
+		Expect(err).NotTo(HaveOccurred())
+
+	})
+
+	ginkgo.AfterEach(func() {
+		if ginkgo.CurrentSpecReport().Failed() {
+			reporter := dump.NewK8sReporter(k8s.FRRK8sNamespace)
+			testName := ginkgo.CurrentSpecReport().FullText()
+			dump.K8sInfo(testName, reporter)
+			dump.BGPInfo(testName, []*frrcontainer.FRR{unnumberedExternalPeer}, cs)
+		}
+	})
+
+	ginkgo.Context("with native neighbor config", func() {
+		ginkgo.It("session should be established and routes to be validated", func() {
+			cr := frrk8sv1beta1.FRRConfiguration{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "native-for-" + unnumberedExternalPeer.Name,
+					Namespace: k8s.FRRK8sNamespace,
+				},
+				Spec: frrk8sv1beta1.FRRConfigurationSpec{
+					BGP: frrk8sv1beta1.BGPConfig{
+						Routers: []frrk8sv1beta1.Router{
+							{
+								ASN: infra.FRRK8sASN,
+								Neighbors: []frrk8sv1beta1.Neighbor{
+									{
+										DynamicASN: frrk8sv1beta1.ExternalASNMode,
+										Interface:  "net0",
+										ToAdvertise: frrk8sv1beta1.Advertise{
```

Comment:
can we also have a receive part and validate it?

Reply:
done